integration-tests/src/lib.rs

```diff
@@ -140,55 +188,153 @@ where
+        if let Some(max_level) = max_level {
+            let filter = match max_level {
+                LevelFilter::TRACE => LogFacilityLevelFilter::Trace,
+                LevelFilter::DEBUG => LogFacilityLevelFilter::Debug,
+                LevelFilter::INFO => LogFacilityLevelFilter::Info,
+                LevelFilter::WARN => LogFacilityLevelFilter::Warn,
+                LevelFilter::ERROR => LogFacilityLevelFilter::Error,
+                LevelFilter::OFF => LogFacilityLevelFilter::Off,
+            };
+            builder = builder.with_max_level(filter);
+        }
+        builder.init().unwrap();
     });
 }
 
-static LOG_ENV_FILTER: Lazy<String> = Lazy::new(|| {
-    env::var_os("XAYN_TEST_LOG")
-        .map(|s| {
-            s.to_str()
-                .expect("XAYN_TEST_LOG must only contain utf-8")
-                .into()
-        })
-        .unwrap_or_else(|| "info,sqlx::query=warn".into())
+fn env_opt_var_os(var: &str) -> Option<String> {
+    env::var(var)
+        .map_or_else(
+            |error| match error {
+                VarError::NotPresent => None,
+                VarError::NotUnicode(err) => panic!("{var} must only contain utf-8: {err:?}"),
+            },
+            Some,
+        )
+        .filter(|v| !v.trim().is_empty())
+}
+
+fn env_var_os_with_default(var: &str, default: &str) -> String {
+    env_opt_var_os(var).unwrap_or_else(|| default.into())
+}
+
+fn select_filter_directives(input: String, default_directives: &str) -> Option<String> {
+    match input.as_str() {
+        "false" => None,
+        "true" => Some(default_directives.to_owned()),
+        _ => Some(input),
+    }
+}
+
+fn directives_with_level_filter(
+    directives: String,
+    var_name: &str,
+) -> (String, Option<LevelFilter>) {
+    let hint = directives
+        .parse::<EnvFilter>()
+        .unwrap_or_else(|err| panic!("{var_name} contains invalid EnvFilter directives: {err}"))
+        .max_level_hint();
+    (directives, hint)
+}
+
+fn additional_env_filter(
+    var_name: &str,
+    default_state: &str,
+    default_value: &str,
+) -> Option<(String, Option<LevelFilter>)> {
+    select_filter_directives(
+        env_var_os_with_default(var_name, default_state),
+        default_value,
+    )
+    .map(|directives| directives_with_level_filter(directives, var_name))
+}
+
+static LOG_ENV_FILTER: Lazy<(String, Option<LevelFilter>)> = Lazy::new(|| {
+    directives_with_level_filter(
+        env_var_os_with_default(XAYN_TEST_LOG, "info,sqlx::query=warn"),
+        XAYN_TEST_LOG,
+    )
 });
 
-pub fn initialize_local_test_logging(_test_id: &str) -> Dispatch {
+static LOG_STDOUT_ENV_FILTER: Lazy<Option<(String, Option<LevelFilter>)>> =
+    Lazy::new(|| additional_env_filter(XAYN_TEST_STDOUT_LOG, "true", &LOG_ENV_FILTER.0));
+
+static LOG_FILE_ENV_FILTER: Lazy<Option<(String, Option<LevelFilter>)>> =
+    Lazy::new(|| additional_env_filter(XAYN_TEST_FILE_LOG, "false", &LOG_ENV_FILTER.0));
```

Comment:
The next PR also adds `LOG_FLAME_ENV_FILTER`, because e.g. you might want to collect `trace` logs for flame graphs or charts.